### PR TITLE
Parsable markup documentation with doxygen

### PR DESCRIPTION
**Pull request by bebehei**
This is a start to have the documentation with doxygen, which also contains markup to be able to compile it at the end of the day.

With this, one's able to compile a document to get an overview of the internal structures of dunst.

I've moved my own comments of the `queues.c` and `log.c` files to be compliant.

### CODE REVIEW

**Comment by bebehei** [comment]
Enforcing  the tags with YCM can get very profitable. See the [GetDocs command](https://github.com/Valloric/YouCompleteMe#the-getdoc-subcommand).

File: HACKING.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
 
+**You can generate an internal overview with doxygen. For this, use `make doc-doxygen` and you'll find an internal overview of all functions and symbols in `docs/html`.**
+
+# Comments
+
+- Comment system is held similar to JavaDoc
+    - Use `@param` to describe all input parameters
+    - Documentation comments should start with a double star (`/**`)
+    - Look forward to properly link all member types
+        - Functions with an appended `()`
```
Comment (tsipinakis) [inline HACKING.md]: These 2 lines needs some more explanation, I can't make much sense of them.

File: HACKING.md [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+- Add comments to all functions and methods
+- Markdown inside the comments is allowed and also desired
+- Add the comments to the prototype. Doxygen will merge the protoype and implementation documentation anyways.
+  Except for **static** methods. Add the documentation header to the implementation and *not to the prototype*.
```
Comment (tsipinakis) [inline HACKING.md]: That period id going to confuse people comma would be more appropriate here.

File: HACKING.md (hunk 1 above)
Comment (bebehei) [reply HACKING.md]: I've added an example and also the way to reference symbols/variables.

**Pull request merged**